packages/map-template/src/components/Map/Map.jsx

```diff
@@ -139,26 +142,28 @@ function Map({ onLocationClick, onVenueChangedOnMap, useMapProviderModule, onMap
     }, [venueOnMap]);
 
     /*
-     * Show the filtered locations on the map based on their IDs or external IDs if present.
-     * Check if the highlight or filter methods exist.
+     * Dynamically filter or highlight location based on the "filteredLocations", "filteredLocationsByExternalIDs" and "hideNonMatches" property.
      */
     useEffect(() => {
-        if (mapsIndoorsInstance) {
-            if (filteredLocations) {
-                if (mapsIndoorsInstance.highlight) {
-                    mapsIndoorsInstance.highlight(filteredLocations.map(location => location.id));
-                } else if (mapsIndoorsInstance.filter) {
-                    mapsIndoorsInstance.filter(filteredLocations.map(location => location.id));
-                }
-            } else if (filteredLocationsByExternalIDs) {
-                if (mapsIndoorsInstance.highlight) {
-                    mapsIndoorsInstance.highlight(filteredLocationsByExternalIDs.map(location => location.id));
-                } else if (mapsIndoorsInstance.filter) {
-                    mapsIndoorsInstance.filter(filteredLocationsByExternalIDs.map(location => location.id));
-                }
+        if (!mapsIndoorsInstance) return;
+
+        // Determine which set of locations to work with
+        // If none of the locations are available, return the function
+        const locations = filteredLocations || filteredLocationsByExternalIDs;
+        if (!locations) return;
+
+        const locationIds = locations.map(location => location.id);
+
+        if (hideNonMatches) {
+            if (mapsIndoorsInstance.filter) {
+                mapsIndoorsInstance.filter(locationIds);
+            }
+        } else {
+            if (mapsIndoorsInstance.highlight) {
+                mapsIndoorsInstance.highlight(locationIds);
```

Comment:
We could fallback to use filter if highlight does not exist.

Reply:
✅